File: 13_Roman_to_Integer.java
class Solution {
    public int convert(char c){
        switch(c){
            case 'I': return 1;
            case 'V': return 5;
            case 'X': return 10;
            case 'L': return 50;
            case 'C': return 100;
            case 'D': return 500;
            case 'M': return 1000;
            default: return 0;
        }
    }
    public int romanToInt(String s) {
        if(s.length() == 1){
            return convert(s.charAt(0));
        }
        int sum = 0;
        for(int i = 0; i < s.length() - 1; i++){
            if(convert(s.charAt(i)) < convert(s.charAt(i + 1))){
                sum += convert(s.charAt(i + 1)) - convert(s.charAt(i));
                i++;
            }
            else{
                sum += convert(s.charAt(i));
            }
        }
        if(convert(s.charAt(s.length() - 2)) >= convert(s.charAt(s.length() - 1))){
            sum += convert(s.charAt(s.length() - 1));
        }
        return sum;
    }
}

File: 13_Roman_to_Integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        m = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000}
        if len(s) == 1:
            return m[s]
        i = 0
        res = 0
        while i < len(s) - 1:
            if m[s[i]] < m[s[i+1]]:
                res += m[s[i+1]] - m[s[i]]
                i += 1
            else:
                res += m[s[i]]
            i += 1
        if m[s[-2]] >= m[s[-1]]:
            res += m[s[-1]]
        return res